assignment1/q2.cpp

```diff
@@ -0,0 +1,33 @@
+struct node{
+    int val;
+    node* next;
+};
+
+node *kth_element(int k, node *head) {
+
+    //number of element on which starts 'k-th to end' fragment
+    int n = length(head) - k;
+    //not sure what to do when n <= 0
+    //return what function found or do sth that shows there was an 'error'?
+
+    int i = 1;  //linked list element no
+    node *ptr = new node;
```

Comment:
When creating a new pointer that traverses on existing data (such as here), you should not allocate new data.
Here you should set ptr to point at existing data:
`node *ptr = head;`

Otherwise, you have memory leaks - you lose the pointer to the new node you created in this line.